clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2487,7 +2487,12 @@ private StoreSetting validateStoreSettingAndPopulate(ControllerClient controller
 
     if (jobSetting.isTargetedRegionPushEnabled && jobSetting.targetedRegions == null) {
       // only override the targeted regions if it is not set and it is a single region push
-      jobSetting.targetedRegions = storeResponse.getStore().getNativeReplicationSourceFabric();
+      // use source grid fabric as target region to reduce data hop, else use default NR source
+      if (!StringUtils.isEmpty(jobSetting.sourceGridFabric)) {
+        jobSetting.targetedRegions = jobSetting.sourceGridFabric;
+      } else {
+        jobSetting.targetedRegions = storeResponse.getStore().getNativeReplicationSourceFabric();
+      }
```

Comment:
Thanks for the fix. My review is a bit late, PR is already merged...This works when user does not specify target region. But if user specifies the target region which is different with source grid fabric, the repush step will still not complete because SOP is missing in source grid fabric. To resolve that, I think we can reset the VeniceWriter before running the repush job.

Reply:
Thanks for your review. Yes if user specifies a region for target colo push, this would not work. The target colo region option when it does not match with source grid would be inefficient way to push data. We can either error in that case or create a writer using the target region. Let me open a PR for it